chain/types/ethtypes/eth_types.go

```diff
@@ -40,18 +40,24 @@ func (e EthUint64) MarshalJSON() ([]byte, error) {
 	return json.Marshal(e.Hex())
 }
 
+// UnmarshalJSON should be able to parse two types of input:
+// 1. a JSON string containing a hex-encoded uint64 starting with 0x
+// 2. a valid string containing a uint64 in decimal
 func (e *EthUint64) UnmarshalJSON(b []byte) error {
 	var s string
-	if err := json.Unmarshal(b, &s); err != nil {
-		return err
-	}
-	parsedInt, err := strconv.ParseUint(strings.Replace(s, "0x", "", -1), 16, 64)
-	if err != nil {
-		return err
+	if err := json.Unmarshal(b, &s); err == nil {
+		parsedInt, err := strconv.ParseUint(strings.Replace(s, "0x", "", -1), 16, 64)
+		if err != nil {
+			return err
+		}
+		eint := EthUint64(parsedInt)
+		*e = eint
+		return nil
+	} else if eint, err := strconv.ParseUint(string(b), 10, 64); err == nil {
+		*e = EthUint64(eint)
+		return nil
 	}
-	eint := EthUint64(parsedInt)
-	*e = eint
-	return nil
+	return fmt.Errorf("cannot parse %s into EthUint64", string(b))
```

Comment:
This message is externally visible, and users don't understand internal types of Lotus.

```suggestion
	return fmt.Errorf("cannot interpret %s as a hex-encoded uint64, or a number", string(b))
```